internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -1261,6 +1261,48 @@ func newImageServiceStatefulSet(ctx context.Context, log logrus.FieldLogger, asc
 			},
 		})
 
+		if asc.spec.ImagePullCAConfigMapRef != nil {
+
+			// Read the referenced ConfigMap to ensure that it contains a single file "extra-ca.crt"
+			additionalCABundleCM := &corev1.ConfigMap{}
+			if err := asc.Client.Get(ctx, types.NamespacedName{Name: asc.spec.ImagePullCAConfigMapRef.Name, Namespace: asc.namespace}, additionalCABundleCM); err != nil {
+				log.WithError(err).Errorf("Failed to get additional CA Bundle config map %s", asc.spec.ImagePullCAConfigMapRef.Name)
+				return err
+			}
+			var err error
+			_, ok := additionalCABundleCM.Data["extra-ca.pem"]
+			if !ok {
+				err = multierror.Append(err, fmt.Errorf("expected to find single file `extra-ca.pem` in the ConfigMap %s but it was not found", asc.spec.ImagePullCAConfigMapRef.Name))
+			}
+			if len(additionalCABundleCM.Data) != 1 {
+				err = multierror.Append(err, fmt.Errorf("found multiple files in ConfigMap %s, a single file was expected but %d files were found", asc.spec.ImagePullCAConfigMapRef.Name, len(additionalCABundleCM.Data)))
+			}
+			if err != nil {
+				return err
+			}
```

Comment:
I feel like all this validation doesn't belong in the StatefulSet mutate function.

I'd say this should be in the `validate` function and we should set a condition if any of the references are invalid.